### PR TITLE
SUS-3118: Use user name lookup in ApiQueryRevisions

### PR DESCRIPTION
**Pull request by mszabo-wikia**
https://wikia-inc.atlassian.net/browse/SUS-3118

### CODE REVIEW

File: includes/api/ApiQueryRevisions.php [hunk 1]
```diff
@@ -348,6 +378,19 @@ public function execute() {
...
+					$userIds[] = $row->rev_user;
+				}
+			}
+			$res->rewind();
+
+			$this->userNames = User::whoAre( $userIds );
+		}
+
 		foreach ( $res as $row ) {
 			if ( ++ $count > $limit ) {
```
Comment (WikiaTech) [inline includes/api/ApiQueryRevisions.php:387]: MAJOR Extract this increment or decrement operator into a dedicated statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS881)

[project-id]: wikia:app

**Comment by WikiaTech** [comment]
SonarQube analysis reported 11 issues
* ![MAJOR][MAJOR] 8 major
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ApiQueryRevisions.php#L81](https://github.com/Wikia/app/blob/79a8938025547486983d3cd0164373a26c4ea8e6/includes/api/ApiQueryRevisions.php#L81): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ApiQueryRevisions.php#L89](https://github.com/Wikia/app/blob/79a8938025547486983d3cd0164373a26c4ea8e6/includes/api/ApiQueryRevisions.php#L89): The Cyclomatic Complexity of this function "execute" is 79 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [ApiQueryRevisions.php#L89](https://github.com/Wikia/app/blob/79a8938025547486983d3cd0164373a26c4ea8e6/includes/api/ApiQueryRevisions.php#L89): This function "execute" has 329 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [ApiQueryRevisions.php#L96](https://github.com/Wikia/app/blob/79a8938025547486983d3cd0164373a26c4ea8e6/includes/api/ApiQueryRevisions.php#L96): Reduce the number of conditional operators (7) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)
1. ![MAJOR][MAJOR] [ApiQueryRevisions.php#L419](https://github.com/Wikia/app/blob/79a8938025547486983d3cd0164373a26c4ea8e6/includes/api/ApiQueryRevisions.php#L419): The Cyclomatic Complexity of this function "extractRowInfo" is 42 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [ApiQueryRevisions.php#L419](https://github.com/Wikia/app/blob/79a8938025547486983d3cd0164373a26c4ea8e6/includes/api/ApiQueryRevisions.php#L419): This function "extractRowInfo" has 158 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [ApiQueryRevisions.php#L515](https://github.com/Wikia/app/blob/79a8938025547486983d3cd0164373a26c4ea8e6/includes/api/ApiQueryRevisions.php#L515): Do not use the global wgParser Parser instance. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWgParserUsage)
1. ![INFO][INFO] [ApiQueryRevisions.php#L398](https://github.com/Wikia/app/blob/79a8938025547486983d3cd0164373a26c4ea8e6/includes/api/ApiQueryRevisions.php#L398): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [ApiQueryRevisions.php#L426](https://github.com/Wikia/app/blob/79a8938025547486983d3cd0164373a26c4ea8e6/includes/api/ApiQueryRevisions.php#L426): Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)
1. ![INFO][INFO] [ApiQueryRevisions.php#L553](https://github.com/Wikia/app/blob/79a8938025547486983d3cd0164373a26c4ea8e6/includes/api/ApiQueryRevisions.php#L553): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[project-id]: wikia:app
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: includes/api/ApiQueryRevisions.php [hunk 2]
```diff
@@ -277,10 +280,37 @@ public function execute() {
...
+
+				if ( empty( $userId ) && !$isAnon ) {
+					$this->dieUsage( 'nonexistent user', 'baduser' );
+				}
+
+				if ( $userId ) {
+					$this->addWhereFld( 'rev_user', $userId );
+				}
+
+				if ( $isAnon ) {
```
Comment (macbre) [inline includes/api/ApiQueryRevisions.php]: Make it an `elseif` ?

**Review by macbre — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 11 issues
* ![MAJOR][MAJOR] 8 major
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ApiQueryRevisions.php#L81](https://github.com/Wikia/app/blob/645dcc72721579db181999e72f0293c7ffbb49a7/includes/api/ApiQueryRevisions.php#L81): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ApiQueryRevisions.php#L89](https://github.com/Wikia/app/blob/645dcc72721579db181999e72f0293c7ffbb49a7/includes/api/ApiQueryRevisions.php#L89): The Cyclomatic Complexity of this function "execute" is 75 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [ApiQueryRevisions.php#L89](https://github.com/Wikia/app/blob/645dcc72721579db181999e72f0293c7ffbb49a7/includes/api/ApiQueryRevisions.php#L89): This function "execute" has 321 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [ApiQueryRevisions.php#L96](https://github.com/Wikia/app/blob/645dcc72721579db181999e72f0293c7ffbb49a7/includes/api/ApiQueryRevisions.php#L96): Reduce the number of conditional operators (7) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)
1. ![MAJOR][MAJOR] [ApiQueryRevisions.php#L411](https://github.com/Wikia/app/blob/645dcc72721579db181999e72f0293c7ffbb49a7/includes/api/ApiQueryRevisions.php#L411): The Cyclomatic Complexity of this function "extractRowInfo" is 42 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [ApiQueryRevisions.php#L411](https://github.com/Wikia/app/blob/645dcc72721579db181999e72f0293c7ffbb49a7/includes/api/ApiQueryRevisions.php#L411): This function "extractRowInfo" has 158 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [ApiQueryRevisions.php#L507](https://github.com/Wikia/app/blob/645dcc72721579db181999e72f0293c7ffbb49a7/includes/api/ApiQueryRevisions.php#L507): Do not use the global wgParser Parser instance. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWgParserUsage)
1. ![INFO][INFO] [ApiQueryRevisions.php#L390](https://github.com/Wikia/app/blob/645dcc72721579db181999e72f0293c7ffbb49a7/includes/api/ApiQueryRevisions.php#L390): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [ApiQueryRevisions.php#L418](https://github.com/Wikia/app/blob/645dcc72721579db181999e72f0293c7ffbb49a7/includes/api/ApiQueryRevisions.php#L418): Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)
1. ![INFO][INFO] [ApiQueryRevisions.php#L545](https://github.com/Wikia/app/blob/645dcc72721579db181999e72f0293c7ffbb49a7/includes/api/ApiQueryRevisions.php#L545): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[project-id]: wikia:app
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**